retrieval/target.go

```diff
@@ -115,10 +115,18 @@ type TargetStatus struct {
 	mu sync.RWMutex
 }
 
+func (ts *TargetStatus) String() string {
```

Comment:
no, it doesn't.... it in fact locks multiple times...

That's a race condition in itself. the LastError call in L119 could yield something else than the call in L120.

It's probably better to lock and take all the information from the fields directly.